src/codemirror/extensions/autocomplete.ts

```diff
@@ -0,0 +1,180 @@
+import {
+  startCompletion,
+  CompletionContext,
+  CompletionResult,
+  Completion,
+  autocompletion,
+} from "@codemirror/autocomplete";
+import { extractEditorStateInfo } from "../utils/extensions-utils";
+import { AutocompletionsMetadata } from "@/types";
+
+function isNewTagContext(lineText: string): boolean {
+  return /<\w*$/.test(lineText);
+}
+
+function generateComponentNameCompletions(
+  partialName: string,
+  options: AutocompletionsMetadata
+): Completion[] {
+  return Object.keys(options)
+    .filter((componentName) =>
+      componentName.toLowerCase().startsWith(partialName.toLowerCase())
+    )
+    .map((componentName) => ({
+      label: componentName,
+      section: "Components",
```

Comment:
Is it typed?

Reply:
it is a title for our own convention, we can name it whatever we want
when typing opening tag I want to always show Components